sparse/unit_test/Test_Sparse_spiluk.hpp

```diff
@@ -278,6 +278,24 @@ void run_test_spiluk_streams(int test_algo, int nstreams) {
   using crsMat_t = CrsMatrix<scalar_t, lno_t, device, void, size_type>;
   using AT       = Kokkos::ArithTraits<scalar_t>;
 
+  // Workaround for OpenMP: skip tests if OMP_NUM_THREADS < nstreams because of
+  // not enough resource to partition
+  bool run_streams_test = true;
+#ifdef KOKKOS_ENABLE_OPENMP
+  if (std::is_same<typename device::execution_space, Kokkos::OpenMP>::value) {
+    const char *env_omp_num_threads = std::getenv("OMP_NUM_THREADS");
+    if (env_omp_num_threads != nullptr) {
```

Comment:
I think you might also need to check the `KOKKOS_NUM_THREADS` as well, in case it is set rather than `OMP_NUM_THREADS`, so for example add 
```
const char *env_kokkos_num_threads = std::getenv("KOKKOS_NUM_THREADS");
```
and an `else if` branch to compare based on that result (something similar but more elegant than my suggestion)